core/heap.h

```diff
@@ -352,6 +352,9 @@ special_heap_init_aligned(uint block_size, uint alignment, bool use_lock, bool e
                           bool persistent, size_t initial_unit_size);
 void
 special_heap_exit(void *special);
+/* special_heap_alloc makes no alignment guarantees beyond the block size given
+ * at special_heap_init.
```

Comment:
I think it's special_heap_init_aligned which needs a comment.  I would also vote for actually aligning instead of this unusual interface of taking in a requested alignment and then not honoring it, which should be a one-line change: setting block_size forward to alignment.

Reply:
Looking at the special allocator - again: It looks like there is a "block_alignment" used in special_heap_create_unit. I am re-visiting this part, it may be an allocator bug if that's supposed to align each block.

Reply:
Yeah, that's what we've been talking about, and I have been suggesting aligning block_size to block_alignment.

Reply:
But looks like it should already do that. It seems if you aligned block_size now, it would cover up what is not working right now (but supposed to work). Let me look at it a little bit more

Reply:
This is what we already discussed offline...

Reply:
Done!